tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/UsingCultureAttribute.cs

```diff
@@ -11,40 +11,42 @@ namespace FakeItEasy.Analyzer.Tests.Helpers
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
     public sealed class UsingCultureAttribute : BeforeAfterTestAttribute
     {
-        private readonly string cultureName;
+#if FEATURE_THREAD_CURRENTCULTURE
+        private static Action<CultureInfo> setCurrentCulture = culture => Thread.CurrentThread.CurrentCulture = culture;
+        private static Action<CultureInfo> setCurrentUiCulture = culture => Thread.CurrentThread.CurrentUICulture = culture;
+        private static Func<CultureInfo> getCurrentCulture = () => Thread.CurrentThread.CurrentCulture;
+        private static Func<CultureInfo> getCurrentUiCulture = () => Thread.CurrentThread.CurrentUICulture;
+#else
+        private static Action<CultureInfo> setCurrentCulture = culture => CultureInfo.CurrentCulture = culture;
+        private static Action<CultureInfo> setCurrentUiCulture = culture => CultureInfo.CurrentUICulture = culture;
+        private static Func<CultureInfo> getCurrentCulture = () => CultureInfo.CurrentCulture;
+        private static Func<CultureInfo> getCurrentUiCulture = () => CultureInfo.CurrentUICulture;
+#endif

```

Comment:
Sorry to nitpick, but why delegates rather than methods?

Reply:
No reason. I can method them.

Reply:
I think it would make the whole thing more readable

Reply:
Methoded.